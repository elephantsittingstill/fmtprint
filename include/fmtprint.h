#ifndef FMTPRINT_H
#define FMTPRINT_H

#include <stdarg.h>

void fmtprint( const char *fstring, ... );

#endif
