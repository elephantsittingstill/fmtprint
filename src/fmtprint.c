#include "internal.h"

void fmtprint( const char *fstring, ... ) {

	va_list args;
	va_start( args, fstring );

#if USE_COMPUTED_GOTO
	parse_format_switch( &fstring, &args );
#else
	parse_format_goto( &fstring, &args );
#endif

	return;
}
