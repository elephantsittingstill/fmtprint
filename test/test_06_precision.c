#include "../include/fmtprint.h"

int main() {
	fmtprint( "The following lines will all have .7 precision:\n"
			  "Int: %.7d\nUnsigned: %.7u\nOctal: %.7o\nHex: %.7x\n"
			  "Float: %.7f\nString: %.7s\nChar: %.7c\n",
			  42069345, 42069345, 42069345, 42069345, 4.20693457,
			  "Hello There!", 'A' );
}
