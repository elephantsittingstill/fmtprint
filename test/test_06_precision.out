The following lines will all have .7 precision:
Int: 42069345
Unsigned: 42069345
Octal: 240366541
Hex: 281ed61
Float: 4.2069345
String: Hello There!
Char: A
